src/main/java/com/artipie/rpm/meta/XmlFile.java

```diff
@@ -0,0 +1,154 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.meta;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stax.StAXSource;
+import javax.xml.transform.stream.StreamResult;
+
+/**
+ * Xml file.
+ *
+ * @since 1.0
+ * @todo #81:30min Introduce an envelope for XMLStreamWriter so that XmlFile
+ *  can extend it and wrap the XMLStreamWriter created in the constructor
+ *  instead of exposing its internal state via the writer() method.
+ * @todo #81:30min Introduce a class named XmlPackagesFile that should be responsible
+ *  of writing the start of the document (as in {XmlFilelists, XmlOthers, XmlPrimary}.startPackages
+ *  and XmlRepomd.begin) by taking this information in its constructor + a XmlFile. It should also
+ *  be responsible of counting the number of packages added. With all this information,
+ *  the alter method can then be moved to XmlPackagesFile and renamed close by exploiting
+ *  directly that information.
+ */
+final class XmlFile {
+
+    /**
+     * XML factory.
+     */
+    private static final XMLOutputFactory FACTORY = XMLOutputFactory.newInstance();
+
+    /**
+     * XML stream.
+     */
+    private final XMLStreamWriter xml;
+
+    /**
+     * XML file path.
+     */
+    private final Path path;
+
+    /**
+     * Primary ctor.
+     * @param path File path
+     */
+    XmlFile(final Path path) {
+        this(path, xmlStreamWriter(path));
+    }
+
+    /**
+     * Primary ctor.
+     * @param path File path
+     * @param xml XML stream writer
+     */
+    private XmlFile(final Path path, final XMLStreamWriter xml) {
+        this.path = path;
+        this.xml = xml;
+    }
+
+    /**
+     * Underlying XML writer.
+     *
+     * @return XML stream writer
+     */
+    public XMLStreamWriter writer() {
+        return this.xml;
+    }
+
+    /**
+     * Override attribute value in the specified tag.
+     *
+     * @param tag Tag to find
+     * @param attribute Attribute to find
+     * @param value Value to be replace
+     * @throws IOException when XML alteration causes error
+     */
+    public void alter(
```

Comment:
@victornoel how about `alterTag` to be more explicit. I believe this should be more readable.